src/test/java/core/basesyntax/service/impl/ParseServiceImplTest.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.OperationType;
+import core.basesyntax.service.ParseService;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class ParseServiceImplTest {
+    private static ParseService parseService;
+
+    @BeforeAll
+    static void setUp() {
+        parseService = new ParseServiceImpl();
+    }
+
+    @Test
+    void parse_validOperation_ok() {
+        List<String> lines = List.of(
+                "b,banana,20",
+                "s,banana,100",
+                "p,banana,5",
+                "r,banana,10"
+        );
+        List<FruitTransaction> expectedList = List.of(
+                new FruitTransaction(OperationType.BALANCE, "banana", 20),
+                new FruitTransaction(OperationType.SUPPLY, "banana", 100),
+                new FruitTransaction(OperationType.PURCHASE, "banana", 5),
+                new FruitTransaction(OperationType.RETURN, "banana", 10)
+        );
+        List<FruitTransaction> actualList = parseService.parse(lines);
+        assertEquals(expectedList, actualList);
+    }
+
+    @Test
+    void parse_invalidOperation_notOk() {
+        List<String> lines = List.of("b,apple,100", "s,apple,20",
+                "p,apple,30", "t,apple,5");
+        assertThrows(RuntimeException.class,
+                () -> parseService.parse(lines));
+    }
+
+    @Test
+    void parse_wrongAmount_notOk() {
```

Comment:
Maybe it's better to call method "invalidAmount" because you have used word "Invalid" in previous and next methods.